lib/mozilla-ast.js

```diff
@@ -532,6 +532,14 @@
                 name: "this",
             });
         },
+        ParenthesizedExpression: function(M) {
+            var node = from_moz(M.expression);
+            if (!node.start.parens) node.start.parens = [];
+            node.start.parens.push(my_start_token(M));
+            if (!node.end.parens) node.end.parens = [];
+            node.end.parens.push(my_end_token(M));
+            return node;
+        },
```

Comment:
> I see you enabled the acorn parentheses option, but how does it correctly work around the issues with annotations before a semicolon or comma?

(Assuming you meant this PR works and just asking to know why...)

Even though `preserveParens` is enabled, we are going to discard the non-standard `ParenthesizedExpression` nodes &minus; but not before recording their positional information.

So rather than range-matching a comment's end-position to the start-position of `AST_Call`, we can perform exact matches instead, with outermost parentheses being the least priority as consistent with our own parser.

Reply:
I see. That and the for loop looking for whitespace only from the comment end to the next node.